### DISCUSSION TITLE
Automatically Start On Boot/Login

### DISCUSSION THREAD

**Discussion by jdholtz** [yuezk/GlobalProtect-openconnect#214] [Q&A] [answered]
Hello. I have been using `globalprotect-openconnect` for eight months now and it has been working great. However, I recently have wanted to automatically connect to my VPN when I connect to the Internet (on boot/login). Is this possible to do? My VPN server requires a login, but maybe there is a way to automatically pass that log-in information to `gpclient` securely?

The closest I have gotten is doing `gpclient --now` which just prompts me to type my password in. However, I have to remember to actually start up the VPN. Is there a built-in way to do this or do I need to set up my own script? I looked a little bit into `gpservice` but I'm not sure what that does (I can't find any documentation on it).

Thanks!

**Comment by yuezk** [comment, marked as answer]
I have the plan to support it. Stay tuned.

**Comment by filoucrackeur** [comment]
I have the same problem It will be great if it was possible de start on pre logon before GDM service for exemple and then if success run gdm with centos rocky alma fedora etc... :)
